server/service/osquery.go

```diff
@@ -1419,10 +1487,16 @@ func (svc *Service) SubmitResultLogs(ctx context.Context, logs []json.RawMessage
 // Query Reports
 ////////////////////////////////////////////////////////////////////////////////
 
-func (svc *Service) SaveResultLogsToQueryReports(ctx context.Context, results []json.RawMessage) {
+func (svc *Service) saveResultLogsToQueryReports(ctx context.Context, unmarshaledResults []*fleet.ScheduledQueryResult, queriesDBData map[string]*fleet.Query) {
```

Comment:
you may want to consider exporting this function to avoid having to test it within `SubmitResultLogs`

Reply:
something like:  https://github.com/fleetdm/fleet/pull/14438